### PR TITLE
[ fix #554 ] efficient comparisons on natural numbers

### LINKED ISSUE #554: Define fast comparison functions based on NATEQUALS and NATLESS builtins
I just remembered @UlfNorell's comment about these after failing to solve the second
part of day 1 of the [advent of code](https://adventofcode.com/2018/) with Agda. You need to use a `Set` and:

* either AVL trees are slow (and use a lot of memory)
* or Integer comparison functions are slow (and use a lot of memory)
* or all of the above

Anyway, I had to switch to Haskell for the second part.

It shouldn't be too much work to define fast (using [`_<_`](https://agda.github.io/agda-stdlib/Agda.Builtin.Nat.html#651)) and cheap (using [`_≤″_`](https://agda.github.io/agda-stdlib/Data.Nat.Base.html#2715)) comparisons

### PR DESCRIPTION
**Pull request by gallais**
Decide on yes/no based on the (fast, builtin) boolean test.

### CODE REVIEW

**Comment by MatthewDaggitt** [comment]
@gallais would there ever be a situation where one would specifically want to use the existing `_≟_` and `_≤″?_` rather than `_≟ᵇ_` and `_≤″ᵇ?_`?

**Comment by gallais** [comment]
If you like having time to grab a coffee while your project typechecks? :D 

Honestly I don't know. Let me try to prove a diagonal principle for both (aka.
`m ?= m` always returns `yes`) to see whether they behave the same or not.

**Comment by gallais** [comment]
The proof was actually nicer for the boolean-based test! :-o
Should I just replace the old versions in favour of the boolean-based ones?

**Comment by MatthewDaggitt** [comment]
> Should I just replace the old versions in favour of the boolean-based ones?

If there's no difference apart from better speed then that sounds like a good idea!

**Comment by gallais** [comment]
Still need to replace `_≤″?_`.

**Comment by MatthewDaggitt** [comment]
> Still need to replace `_≤″?_`.

That'd be great. In the interests of documentation can you also leave a short comment by the definitions explaining why we're proving them via the boolean version.

File: src/Data/Nat/Properties.agda [hunk 1]
```diff
@@ -42,15 +44,33 @@ open ≡-Reasoning
...
-suc m ≟ suc n  with m ≟ n
-... | yes refl = yes refl
-... | no m≢n   = no (m≢n ∘ suc-injective)
+m ≟ n = map′ (≡ᵇ⇒≡ m n) (≡⇒≡ᵇ m n) (T? (m ≡ᵇ n))
+
+≡-irrelevant : Irrelevant {A = ℕ} _≡_
+≡-irrelevant = Decidable⇒UIP.≡-irrelevant _≟_
+
+≟-diag : ∀ {m n} (eq : m ≡ n) → (m ≟ n) ≡ yes eq
+≟-diag = ≡-≟-identity _≟_
```
Comment (MatthewDaggitt) [inline src/Data/Nat/Properties.agda]: Just noticed this proof. I'm not sure this one should be here? There's nothing particular to natural numbers in it.

Comment (gallais) [reply src/Data/Nat/Properties.agda]: I'd argue it's good to have a specialised version available.

**Pull request merged**